homeassistant/components/nws/weather.py

```diff
@@ -165,85 +73,73 @@ def convert_condition(time, weather):
     return cond, max(prec_probs)
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities, discovery_info):
     """Set up the NWS weather platform."""
-
     latitude = config.get(CONF_LATITUDE, hass.config.latitude)
```

Comment:
We normally add a guard clause here that checks if discovery_info is None and return if so. This can happen if a user configures the platform config section by mistake.